ompi/debuggers/ompi_msgq_dll.c

```diff
@@ -1157,8 +1157,18 @@ static int fetch_request( mqs_process *proc, mpi_process_info *p_info,
         mqs_fetch_data( proc, ompi_datatype + i_info->ompi_datatype_t.offset.name,
                         64, data_name );
         if( '\0' != data_name[0] ) {
-            snprintf( (char*)res->extra_text[1], 64, "Data: %d * %s",
-                      (int)res->desired_length, data_name );
+            // res->extra_text[x] is only 64 chars long -- same as
```

Comment:
This changes the semantics of the returned string to please some unreasonably picky compilers. These warnings should be ignored. In most case, there will be NO truncation, because the longest predefined type is 14 chars, aka 'double complex' and the count cannot be larger than 20 chars (because its value is smaller than 2^64). The only case where information could be lost, is when the user named their datatypes with a long name. Even in that case, it does not matter if the string gets truncated, because this is optional information. However, tools that parse this string will get confused by the change in semantics.

Reply:
I don't disagree that the compiler is being overly-picky, and we'll almost never see an actual truncation in practice. 
 That being said:

1. I do tire of seeing this warning.
1. The array has 5 lines we can use for exactly this kind of reason.
1. Do tools actually search for this specific string?
1. This is a cherry pick from master; we've had this on master (and v5.0.x) for quite a while.  That being said, neither of those is released...